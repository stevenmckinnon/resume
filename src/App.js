import React from 'react';
import './App.css';
import Header from './components/header/Header';
import About from './components/about/About';
import Resume from './components/resume/Resume';
import Footer from './components/footer/Footer';
import { graphql } from 'react-apollo';
import personalDetails from './queries/query';

const App = props => {
    if (props.data.error) {
        return <h1>Error fetching the data!</h1>;
    } else if (!props.data.loading) {
        const personalDetails = props.data.personalDetailses.edges[0].node;
        const {
            workExperiences,
            educations,
            intro,
            socialMedia,
            name,
            photo
        } = personalDetails;
        const { skills } = personalDetails.skills;

        return (
            <div className="App">
                <Header name={name} intro={intro} socialMedia={socialMedia} />
                <About personalDetails={personalDetails} photo={photo} />
                <Resume work={workExperiences} education={educations} skills={skills} />
                <Footer name={name} socialMedia={socialMedia} />
            </div>
        );
    }

    return (
        <div className="loading">
            <div id="spinContain" />
            <div id="spinContainT" />
        </div>
    );
};

export const AppComp = App;
export default graphql(personalDetails, {
    options: () => ({
        variables: {
            name: 'Steve McKinnon'
        }
    })
})(App);