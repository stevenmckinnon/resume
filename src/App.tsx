/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import * as React from "react"
import "./App.css"
import "./index.css"

import Main from "./sections/main/Main"
import About from "./sections/about/About"
import Resume from "./sections/resume/Resume"
import Footer from "./sections/footer/Footer"
import { useQuery } from "@apollo/client"
import personalDetails from "./queries/query"
import { Header } from "@/components/Header"

const App: React.FC = () => {
  const data = useQuery(personalDetails, {
    variables: { name: "Steve McKinnon" },
  })

  if (data.error != null) {
    return <h1>Error fetching the data!</h1>
  } else if (!data.loading) {
    const personalDetails = data.data.personalDetailses.edges[0].node
    const { workExperiences, educations, intro, socialMedia, name, photo } =
      personalDetails
    const { skills } = personalDetails.skills

    return (
      <div className="App z-0 overflow-x-hidden">
        <Header />
        <Main name={name} intro={intro} socialMedia={socialMedia} />
        <About personalDetails={personalDetails} photo={photo} />
        <Resume work={workExperiences} education={educations} skills={skills} />
        <Footer name={name} socialMedia={socialMedia} />
      </div>
    )
  }

  return (
    <div className="loading">
      <div id="spinContain" />
      <div id="spinContainT" />
    </div>
  )
}

export const AppComp = App
export default App
