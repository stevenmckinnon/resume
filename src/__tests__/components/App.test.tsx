import { shallow } from 'enzyme';
import * as React from 'react';
import { AppComp } from '../../App';

describe('Tests the App component', () => {
    it('renders the App component loading', () => {
        const data = {
            loading: true,
            error: null,
            personalDetailses: {}
        };
        const wrapper = shallow(<AppComp data={data} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders the App component error', () => {
        const data = {
            loading: false,
            error: 'error',
            personalDetailses: {}
        };
        const wrapper = shallow(<AppComp data={data} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders the App component data', () => {
        const data = {
            loading: false,
            error: null,
            personalDetailses: {
                edges: [{
                    node: {
                        name: 'John Smith',
                        location: 'New York, NY',
                        biography: 'I am a dev, hear me roar',
                        cvUrl: 'http://url.com',
                        email: 'hello@email.com',
                        workExperiences: [
                            {
                                company: 'Company Name',
                                description: 'Description of role',
                                fromDate: '2018-06-20T00:00:00.000Z',
                                toDate: null,
                                jobTitle: 'Job title 1'
                            },
                            {
                                company: 'Company Name 2',
                                description: 'Description of role',
                                fromDate: '2018-06-20T00:00:00.000Z',
                                toDate: null,
                                jobTitle: 'Job title 2'
                            },
                        ],
                        educations: [{
                            name: 'Uni name',
                            course: 'Course name (BSc Hons)',
                            fromDate: '2010-09-13T00:00:00.000Z',
                            toDate: '2014-05-21T00:00:00.000Z'
                        }],
                        skills: [
                            'JavaScript',
                            'HTML',
                            'CSS',
                            'Partying'
                        ],
                        socialMedia: {
                            twitter: 'http://twitter.com/stevenmckinnon',
                            linkedIn: 'https://www.linkedin.com/in/stevenmckinnon92/',
                            instagram: 'http://instagram.com/stevenmckinnon',
                            photography: 'http://stevenmckinnon.co.uk',
                            github: 'https://github.com/stevenmckinnon'
                        },
                        photo: {
                            handle: 'handle',
                            width: 1333,
                            height: 1333
                        }
                    }
                }]
            }
        };
        const wrapper = shallow(<AppComp data={data} />);
        expect(wrapper).toMatchSnapshot();
    });
});
