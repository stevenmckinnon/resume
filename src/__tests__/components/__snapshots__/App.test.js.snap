// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the App component renders the App component data 1`] = `ShallowWrapper {}`;

exports[`Tests the App component renders the App component error 1`] = `ShallowWrapper {}`;

exports[`Tests the App component renders the App component loading 1`] = `ShallowWrapper {}`;
